graylog2-web-interface/src/components/common/Spinner.jsx

```diff
@@ -1,22 +1,52 @@
-import React from 'react';
+// @flow
+import React, { useState, useEffect } from 'react';
+import type { ComponentType } from 'react';
 import PropTypes from 'prop-types';
+import styled from 'styled-components';
 
 import Icon from './Icon';
 
+const Wrapper: ComponentType<{ visible: boolean }> = styled.span`
+  visibility: ${({ visible }) => (visible ? 'visible' : 'hidden')};
```

Comment:
Should this be `display: none`? `visibility` keeps the space in the layout allocated while display removes the rendering completely